mitodl/odl-video-service
# Update youtube oauth (and python, dependencies)

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes #989 

#### What's this PR do?
- Updates oauth libraries, the methods used to generate youtube tokens, and youtube authentication.  Also updates python and many other python modules.  Saving javascript updates for another PR.

#### How should this be manually tested?
- Unit tests should pass
- You will need to get a bunch of .env variables from me or Sar. 
- Run ngrok and point it to your local OVS: `./ngrok http 8089`
- Create a superuser
- Log in to OVS via the ngrok URL
- Go to `./api/v0/youtube-tokens/`, you should be redirected to a Google auth page.  Follow the prompts and eventually you should be redirected back to OVS with a JSON response containing values for `YT_ACCESS_TOKEN` and `YT_REFRESH_TOKEN`
- Copy those values into your `.env` file, then restart containers
- Create a collection, then upload a small video.  After a few minutes (if it's small enough) your video should be playable.
- In django admin, find your video and set `Is public` to `True`
- In a shell, run `from cloudsync.tasks import *;upload_youtube_videos.delay()`
- Within a few minutes your video should be available on Youtube (check under django admin, `Youtube videos` section).


#### Any background context you want to provide?
The easiest way to review the code is by individual commit; most files were modified in one commit and just contain formatting/linting changes.
